src/einsteinpy/symbolic/weyl.py

```diff
@@ -0,0 +1,96 @@
+import numpy as np
+import sympy
+
+from einsteinpy.symbolic.ricci import RicciScalar, RicciTensor
+from einsteinpy.symbolic.riemann import RiemannCurvatureTensor
+from einsteinpy.symbolic.tensor import Tensor
+
+
+class WeylTensor(Tensor):
+    """
+    Class for defining Weyl Tensor
+    """
+
+    def __init__(self, arr, syms, config="llll", parent_metric=None):
+        """
+        Constructor and Initializer
+
+        Parameters
+        ----------
+        arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+            Sympy Array or multi-dimensional list containing Sympy Expressions
+        syms : tuple or list
+            Tuple of crucial symbols dentoting time-axis, 1st, 2nd, and 3rd axis (t,x1,x2,x3)
+        config : str
+            Configuration of contravariant and covariant indices in tensor. 'u' for upper and 'l' for lower indices. Defaults to 'llll'.
+        parent_metric : ~einsteinpy.symbolic.metric.WeylTensor
+            Corresponding Metric for the Weyl Tensor. Defaults to None.
+
+        Raises
+        ------
+        TypeError
+            Raised when arr is not a list or sympy Array
+        TypeError
+            syms is not a list or tuple
+        ValueError
+            config has more or less than 3 indices
+
+        """
+        super(WeylTensor, self).__init__(arr, config=config)
+        self.syms = syms
+        self._order = 4
+        self._parent_metric = parent_metric
+        if isinstance(syms, (list, tuple)):
+            self.syms = syms
+            self.dims = len(self.syms)
+        else:
+            raise TypeError("syms should be a list or tuple")
+        if not len(config) == self._order:
+            raise ValueError("config should be of length {}".format(self._order))
+
+    @classmethod
+    def from_metric(cls, metric, dims, parent_metric=None):
+        """
+        Get Weyl tensor calculated from a metric tensor
+
+        Parameters
+        ----------
+        metric : ~einsteinpy.symbolic.metric.MetricTensor
+            Space-time Metric from which Christoffel Symbols are to be calculated
+        dims : int
+            Dimension of Weyl Tensor
+        parent_metric : ~einsteinpy.symbolic.metric.WeylTensor
+            Corresponding Metric for the Weyl Tensor. Defaults to None
+
+        Raises
+        ------
+        TypeError
+            Raised when the dimension of the tensor is less than 3
+
+        """
+        if dims >= 3:
+            if dims > 3:
+                t_riemann = RiemannCurvatureTensor.from_metric(metric)
+                t_ricci = RicciTensor.from_metric(metric)
+                r_scalar = RicciScalar.from_riccitensor(t_ricci, parent_metric)
+                aux = sympy.tensorproduct(t_ricci, metric)
+                interm = sympy.tensorproduct(metric, metric)
```

Comment:
```
aux = sympy.tensorproduct(t_ricci, metric)
aux = np.array(aux).reshape(aux.shape)

interm = sympy.tensorproduct(metric, metric)
interm = np.array(interm).reshape(interm.shape)
```

You have to convert the tensors to numpy array before applying `moveaxis` . See https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.moveaxis.html

Also find a better name than `interm`   xD